### PR TITLE
Fixed Issues hopefully

### PR DESCRIPTION
**Pull request by mytopdog**

### CODE REVIEW

File: src/Classes/Invite.js [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
 
-        this.channel = raw.channel;
-    }
+    this.channel = raw.channel;
+  }
+
+  delete() {
+    return new Promise((res, rej) => {
+      request.req('DELETE', `/invites/${this.code}`, {}, this.client.token).then(i => {
+        var invite = new Invite(i);
```
Comment (YoussefElshemi) [inline src/Classes/Invite.js]: You forget to pass the client object to the class

Comment (mytopdog) [reply src/Classes/Invite.js]: oh rip

Comment (mytopdog) [reply src/Classes/Invite.js]: will fix

Comment (mytopdog) [reply src/Classes/Invite.js]: Fixed :+1:

File: src/Classes/Guild.js [hunk 2]
```diff
@@ -170,7 +171,18 @@ class Guild {
...
         topic: (type === 'text' && opt && opt.topic) || null
       }, this.client.token).then(c => {
-        setTimeout(res, 100, res(this.client.channel_methods().fromRaw(c)));
+        const TextChannel = require('./TextChannel');
+        const VoiceChannel = require('./VoiceChannel');
+
+        let ret;
+
+        if (c instanceof TextChannel) {
+          ret = new TextChannel(this.client.channel_methods().fromRaw(c), this.client);
```
Comment (YoussefElshemi) [inline src/Classes/Guild.js]: these could be one lines

File: src/Classes/Guild.js [hunk 3]
```diff
@@ -170,7 +171,18 @@ class Guild {
         nsfw: (type === 'text' && opt && opt.nsfw) || null,
         topic: (type === 'text' && opt && opt.topic) || null
       }, this.client.token).then(c => {
-        setTimeout(res, 100, res(this.client.channel_methods().fromRaw(c)));
+        const TextChannel = require('./TextChannel');
```
Comment (YoussefElshemi) [inline src/Classes/Guild.js]: So you don't require multiple times, require outside of the class

Comment (mytopdog) [reply src/Classes/Guild.js]: Can't do that, it causes error:
`Class extends value #<Object> is not a constructor or null`

File: src/Classes/Guild.js (hunk 2 above)
Comment (mytopdog) [reply src/Classes/Guild.js]: Possibly I'll make a new commit doing that soon

File: src/Classes/Guild.js (hunk 3 above)
Comment (YoussefElshemi) [reply src/Classes/Guild.js]: Ok

File: src/Classes/Guild.js (hunk 2 above)
Comment (mytopdog) [reply src/Classes/Guild.js]: Ok done now :+1:

**Comment by mytopdog** [comment]
```js
Client.on('message', message => {
  if (message.content.startsWith('!create')) {
    message.guild.createChannel(message.content.split(' ')[1], 'text').then(c => {
      c.send('Created #' + message.content.split(' ')[1]);
    }).catch(console.error);
  }
});
```
Code to test for issue regarding channels not caching guilds (I think, that's what I've gathered so far)

**Pull request merged**